### PR TITLE
PaletteEditor update

### PR DESCRIPTION
**Pull request by y4my4my4m**
- [x] Fix unecessary use of 32bit color
  - [x] Uses GrRect instead of DrawPixel
- [x] Added variants to PaletteGod
  - [x] Clean up
  - [x] Base function by @doodayev 
  - [x] Added randomizer for a single color (within PaletteEditor too)
  - [x] Added randomizer for whole palette (within PaletteEditor too)
 - [x] New Palette `Tradition` that properly uses the CGA colors (red is red, blue is blue, etc)

### CODE REVIEW

**Comment by y4my4my4m** [comment]
@doodayev currently i've "fixed" the 32bit colors to 16bit colors, but it crashes when playing with it/changing the window size. Not sure what causes it. Can you look into it?

**Comment by doodayev** [comment]
Not to nitpick but you probably mean 4-bit color since 16-bit colors means 65,536 colors. Anyways sure I'll try to see what's up with the Palette Editor.

**Comment by y4my4my4m** [comment]
@doodayev i meant 16colors lol

**Comment by y4my4my4m** [comment]
Just a small update, not a major overhaul. What changed is in the PR description.

**Review by doodayev — DISMISSED** [review]
Nice work! Everything here looks good to me!

File: src/Home/PaletteEditor/PaletteEdit.ZC [hunk 1]
```diff
@@ -38,13 +38,12 @@ U0 DrawCtrlSlider(CDC *dc, CCtrl *c)
 
 	CSliderState *s = c->state;
 	I64	size = (c->win_task->pix_width / 16),
-		border = size / 16,
+//		border = size / 16,
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteEdit.ZC:41]: Why not delete the line entirely?

File: src/Home/PaletteEditor/PaletteEdit.ZC [hunk 2]
```diff
@@ -73,6 +63,9 @@ U0 DrawCtrlSlider(CDC *dc, CCtrl *c)
 	GrRect(dc, 94, 8, 58, 12);
 
 	dc->color = selected_color;
+	// Color preview next to #hex
+	GrRect(dc, 78, 8, 12, 12);
+	// #Hex value
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteEdit.ZC:68]: I find this comment unneceary, can be easily understood from the lines below.

File: src/Home/PaletteEditor/PaletteEdit.ZC [hunk 3]
```diff
@@ -183,23 +159,34 @@ U0 DrawCtrlSlider(CDC *dc, CCtrl *c)
...
+		GrPrint(dc, bias + (i * size), c->win_task->pix_height - (size * .6) + (size / 4), "%d", i);
+		GrPrint(dc, 10, (size2 * 2) + 70 + 16 * i, "#%d: Test", i);
+		// Border for preview box
+		GrRect(dc, (size * i)-2, c->win_task->pix_height - (size*1.5)-1, size+2, size+1);
+		// preview box
+		dc->color = i;
+		GrRect(dc, (size * i), c->win_task->pix_height - (size*1.5), size, size);
+
+
+
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteEdit.ZC:189]: Why those empty lines?

File: src/Home/PaletteEditor/PaletteGod.ZC [hunk 4]
```diff
@@ -1,15 +1,23 @@
+CBGR24 ColorGod(Bool instant=TRUE)
+{
+ 	// Generate a random color
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteGod.ZC:3]: Put the comment on the same line as the `{`.

File: src/Home/PaletteEditor/PaletteGod.ZC [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
+ 	if (instant)
+ 		return RandI32 & 0xFFFFFF;
+ 	else
+ 		return GodPick & 0xFFFFFF;
+}
 
-public U0 PaletteGodGenerate()
+public U0 PaletteGod(Bool seedUse=FALSE, Bool instant=TRUE)
 {
+ 	// Generate a random palette, seed-based or not
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteGod.ZC:12]: Same here, put the comment on the same line.

File: src/Home/PaletteEditor/PaletteGod.ZC [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
+		Seed(GodPick);
 	}
+ 	for (i=0; i < 16; i++)
+ 		gr_palette[i] = ColorGod(instant);
+	LFBFlush;
 }
 
-PaletteGodGenerate;
\ No newline at end of file
+// PaletteGod;
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteGod.ZC:23]: Remove this comment if unused.

File: src/Home/PaletteEditor/Palettes/Tradition.ZC [hunk 7]
```diff
@@ -0,0 +1,10 @@
+public CBGR24 gr_palette_tradition[COLORS_NUM] = {
+0xE8E8E9, 0x374B88, 0x4FBC2C, 0x29CACE, 0xDD0000, 0x8C33FF, 0x875721, 0x7F7A7C, 0x15171C, 0x2EA1FF, 0x4FFF00, 0xFFBB, 0x690000, 0x6E3FFC, 0xFFCC00, 0xC0F14
+};
+public U0 PaletteSetTradition(Bool persistent=TRUE)
+{
+	GrPaletteSet(gr_palette_tradition);
+	LFBFlush;
+	if (persistent)
+		fp_set_std_palette = &PaletteSetTradition;
+}
```
Comment (xslendix) [inline src/Home/PaletteEditor/Palettes/Tradition.ZC:10]: You may have a cursor position character here.

File: src/Home/PaletteEditor/PaletteEdit.ZC [hunk 8]
```diff
@@ -152,27 +145,10 @@ U0 DrawCtrlSlider(CDC *dc, CCtrl *c)
...
-					else
-						tempColor = 0x000000;
-				}
-				// color preview
-				DrawPixel(c->win_task->pix_left + j + (size * i), c->win_task->pix_bottom - size * 1.5 + k, tempColor);
-			}
-		}
 		// color label and text preview
 		// TODO: THIS LOGIC DOESNT WORK ON LIGHT THEMES		
 		// if color too light or too dark add bg
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteEdit.ZC:150]: I would personally use proper punctuation for those comments. Even if they maybe are temporary.

File: src/Home/PaletteEditor/PaletteEdit.ZC [hunk 9]
```diff
@@ -38,13 +38,12 @@ U0 DrawCtrlSlider(CDC *dc, CCtrl *c)
 
 	CSliderState *s = c->state;
 	I64	size = (c->win_task->pix_width / 16),
-		border = size / 16,
+//		border = size / 16,
 		size2 = (c->win_task->pix_height / 32),
 //		slider_amount = 3,
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteEdit.ZC:43]: This can also be deleted.

**Review by xslendix — CHANGES_REQUESTED** [review]
Overall, code tooks good to me, but the formatting of the code needs some work so it's unified with the rest of the system. Some comments need some work or can be removed entirely. Just look at my comments under the review I have provided.

File: src/Home/PaletteEditor/PaletteGod.ZC [hunk 10]
```diff
@@ -1,15 +1,23 @@
...
-	CBGR24	palette;
 
-	for (i = 0; i < 16; i++)
-	{
-		palette = GodPick & 0xFFFFFF;
-		gr_palette[i] = palette;
-		LFBFlush;
+	if (seedUse) {
+		Seed(GodPick);
 	}
```
Comment (xslendix) [inline src/Home/PaletteEditor/PaletteGod.ZC:17]: You can make this 2 lines instead of 3.

Comment (y4my4my4m) [reply src/Home/PaletteEditor/PaletteGod.ZC:17]: @xslendix you can actually make this one line trollface.jpg
`if (seedUse) Seed(GodPick);` 

Thanks for the review, I'll work on it this weekend.

File: src/Home/PaletteEditor/PaletteEdit.ZC (hunk 8 above)
Comment (y4my4my4m) [reply src/Home/PaletteEditor/PaletteEdit.ZC:150]: unchanged, wont fix as there's plenty that needs rework in paletteedit

**Comment by xslendix** [comment]
LGTM, but I have one more gripe: comments that are next to { usually are something like this:
```holyc
U0 ....
{// My documentation
```
Not
```holyc
U0 ....
{       // My documentation
```

**Comment by y4my4my4m** [comment]
@xslendix @Midnoclose 
re-review please

**Review by xslendix — APPROVED** [review]

**Review by TempleProgramming — APPROVED** [review]

**Pull request merged**